### DISCUSSION TITLE
Static IP Address for sonoff device

### DISCUSSION THREAD

**Discussion by Dant009** [RavenSystem/esp-homekit-devices#2180] [New Features]
It’s a feature that can prevent some ip conflict, and can gets better manage of all the devices

**Comment by RavenSystem** [comment]
DHCP is the service that can prevent some IP conflicts.

**Reply by Dant009** [reply]
I have a router that once it assign an ip to a device, it continue to assign the same ip to the same device, like it’s has a “ip table”. The problem is if the sonoff has the ip 192.168.0.100, and my phone too (in the ip table of the router) when I reconnect the phone to the router, dhcp doesn’t work and assign the same ip to 2 device, and or the sonoff or the phone work in the network. 
So I can start the DHCP of my router above the .100 and set all the static ip from 0 to 99, for example. 
thank you for all this project!

**Reply by RavenSystem** [reply]
If DHCP server of your router assigns same IP address to two different devices when both are connected, means that your router has a very big bug. Wasting a lot of time and resources to add static IP address support to HAA is not the solution.

**Reply by Dant009** [reply]
Ok I understand that. 
thank you so much!

**Reply by RavenSystem** [reply]
If you have a RPi, or a NAS, you can install there a DHCP server and disable DHCP server of your router. You can check too if your router has a firmware upgrade that fixes it, or ask manufacturer about fixing that bug, or change your router.

**Reply by Dant009** [reply]
Perfect! Thanks for the idea. I’ll try

**Reply by RavenSystem** [reply]
The NAS is the best option. Normally, QNAP and Synology have a very good built-in DHCP server, managed via web UI. But if you use a RPi, dnsmasq is a great DHCP server. And don't forget to disable DHCP server from router to avoid issues.
